cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/graph/Extensions.kt

```diff
@@ -85,6 +88,24 @@ inline fun <reified T> Node?.allChildrenWithOverlays(
     }
 }
 
+/**
+ * Searches for all Overlay nodes that are connected to a node in this nodes sub AST and returns all
+ * nodes of type [T]. For convenience, an optional predicate function [predicate] can be supplied,
+ * which will be applied via [Collection.filter].
+ */
+@JvmOverloads
+inline fun <reified T> Node?.allOverlays(noinline predicate: ((T) -> Boolean)? = null): List<T> {
```

Comment:
There is already `allChildrenWithOverlays` which I think does essentially the same thing.

Reply:
Removed this function